src/infrastructure/repositories/ViemChainMetricsRepository.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  createPublicClient,
+  http,
+  PublicClient
+} from "viem"
+// @ts-ignore
+import { hemiSepolia, hemi } from "hemi-viem"
+// @ts-ignore
+import { Hash } from '../../domain/valueObjects/Hash'
+import { ChainMetricsRepository } from '../../domain/repositories/ChainMetricsRepository'
+import { ChainMetrics } from '../../domain/entities/ChainMetrics'
+import { BlockNumber } from '../../domain/valueObjects/BlockNumber'
+import { Metric } from '../../domain/valueObjects/Metric'
+
+export class ViemChainMetricsRepository implements ChainMetricsRepository {
+  private client: PublicClient
+  private blockExplorerUrl: string
+
+  constructor() {
+    const chain = !!process.env['TESTNET'] ? hemiSepolia : hemi
+    
+    // @ts-expect-error
+    this.client = createPublicClient({ chain, transport: http() })
+    this.blockExplorerUrl = process.env['BLOCK_EXPLORER_URL'] || ''
+  }
+
+  async collect(): Promise<ChainMetrics> {
+    const blockNumber = await this.client.getBlockNumber()
+    
+    const response = await fetch(`${this.blockExplorerUrl}/api/v2/stats`)
```

Comment:
It seems these 2 requests are unrelated, and can be run in parallel and save some time, instead of running them in sequence

```suggestion
    const [blockNumber, response] = await Promise.all([this.client.getBlockNumber(), fetch(`${this.blockExplorerUrl}/api/v2/stats`)])
```

Reply:
fixed